SSA/Experimental/IntrinsicAsymptotics.lean

```diff
@@ -88,57 +88,54 @@ theorem ICom.denote_changeVars {Γ Γ' : Ctxt}
     funext t v
     cases v using Ctxt.Var.casesOn <;> simp
 
-/-- Append two programs, while substituting a free variable in the second for 
-the output of the first -/
-def addProgramAtTop {Γ Γ' : Ctxt} (v : Γ'.Var t₁)
-    (map : Γ.Hom Γ') :
-    (rhs : ICom Γ t₁) → (inputProg : ICom Γ' t₂) → ICom Γ' t₂
-  | .ret e, inputProg => inputProg.changeVars 
-      (fun t' v' => 
-        if h : ∃ h : t₁ = t', h ▸ v = v' 
-        then h.fst ▸ map e
-        else v')
-  | .lete e body, inputProg => 
-      let newBody := addProgramAtTop v.toSnoc
-        (fun _ v => Ctxt.Hom.snocMap map v)
-        body 
-        (inputProg.changeVars (fun _ v => v.toSnoc))
-      .lete (e.changeVars map) newBody
-      
-theorem denote_addProgramAtTop {Γ Γ' : Ctxt} (v : Γ'.Var t₁)
-    (map : Γ.Hom Γ') (s : Γ'.Valuation) :
-    (rhs : ICom Γ t₁) → (inputProg : ICom Γ' t₂) → 
-    (addProgramAtTop v map rhs inputProg).denote s =
-      inputProg.denote (fun t' v' => 
-        if h : ∃ h : t₁ = t', h ▸ v = v' 
-        then h.fst ▸ rhs.denote (fun t' v' => s (map v'))
-        else s v')
-  | .ret e, inputProg => by
-    simp only [addProgramAtTop, ICom.denote_changeVars, ICom.denote]
+structure addProgramToLets.Result (Γ_in Γ_out : Ctxt) (ty : Ty) where
+  {Γ_out_new : Ctxt}
+  lets : Lets Γ_in Γ_out_new
+  diff : Ctxt.Diff Γ_out Γ_out_new
+  var : Γ_out_new.Var ty
+
+/--
```

Comment:
Maybe the documentation below should go here to explain the result? (I don't feel strongly about this at all, just a thought)